### PR TITLE
_user/voxeldemo: Add double buffering and make run on "any" graph video adapter

### PR DESCRIPTION
**Pull request by gerard5**
RTOS: 57

## Description

This PR adds double buffering to fix fickering on Cirrus adapter and enables to run demo on "any" graph video adapter which support 32bit true color.

## Motivation and Context

Fixes flickering on cirrus.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (ia32 with and without kvm hwaccel, using virtiogpu as well cirrus qemu adapter).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: _user/voxeldemo/main.c [hunk 1]
```diff
@@ -81,14 +84,19 @@ int voxel_init(voxel_t *v)
...
+			break;
+
+		v->width = g->width;
+		v->height = g->height;
 
-		return 0;
+		return EOK;
 	} while (0);
 
 	voxel_free(v);
```
Comment (kemonats) [inline _user/voxeldemo/main.c:102]: What value will have the `v->mapcolor variable`
if  `(v->mapHeight = malloc(1024 * 1024)) == NULL` will be true?
You should set variables before allocating memory.
```
v->mapColor = NULL;
v->bufback = NULL;
v->pixels = NULL;
v->palette = NULL;
```

Comment (gerard5) [reply _user/voxeldemo/main.c:102]: It is expected to set `voxel_t` structure/context properly before `voxel_init(..)` is called, thus:
```
	voxel_t v = {
		.cam = { .h = 300, .horiz = 200, .dist = 300 },
	};
```
is enough, any other member of the `voxel_t` structure is/are set to `NULL`, `0` by default, so all `mapHeight`, `mapColor`, `backBuf`, `pixels` and `palette` are set to `NULL` by default, (this is property of C initializer), before call to `voxel_init(..)`.

So, to answer Your question `mapHeight` will be `NULL`, but `mapColor` will be assigned a valid pointer returned by successful `malloc`, later `voxel_free(v)` is called and function returns an error, program fail then, and exits.

**Comment by pawelpisarczyk** [comment]
Looks good to me./

**Review by pawelpisarczyk — APPROVED** [review]
Looks good to me.

**Pull request merged**